pysmt/solvers/btor.py

```diff
@@ -230,9 +231,11 @@ def get_model(self):
 
     @clear_pending_pop
     def _solve(self, assumptions=None):
+        self._last_assumptions = []
         if assumptions is not None:
             btor_assumptions = [self.converter.convert(a) for a in assumptions]
             self.btor.Assume(*btor_assumptions)
+            self._last_assumptions = list(assumptions)
```

Comment:
If assumption is an iterator, we will consume it in L234, and then L238 will return an empty list.
Shall we do `list(assumptions)` earlier?